test/unit/toggle/toggle-component.js

```diff
@@ -0,0 +1,151 @@
+import Toggle from '../../../src/components/toggle';
+import ToggleView from '../../../src/views/toggle';
+
+const fakeClient = {
+  fetchCart: () => { return; },
+  createCart: () => { return; },
+};
+
+describe('Toggle Component class', () => {
+  describe('constructor', () => {
+    let toggle;
+    let cart;
+
+    beforeEach(() => {
+      cart = {
+        node: document.createElement('div'),
+      };
+      document.body.appendChild(cart.node);
+    });
+
+    afterEach(() => {
+      document.body.removeChild(cart.node);
+    });
+
+    it('sets typeKey to toggle', () => {
+      toggle = new Toggle({}, {client: fakeClient, cart});
+      assert.equal(toggle.typeKey, 'toggle');
+    });
+
+    it('sets node to node in config if provided', () => {
+      const node = {};
```

Comment:
Could we set this to a `document.createElement` to more closely mimic a real node?